controllers/alertsconfig_controller.go

```diff
@@ -82,6 +82,7 @@ func (r *AlertsConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	// Get the CR
 	var alertsConfig alertmanagerv1alpha1.AlertsConfig
+	alertsConfig.Status.RetryCount = 0
```

Comment:
We cannot do this because it will always set the retryCount to be 0 even for error retry case. I think you need set retryCount to 0 for successful case [create](https://github.com/keikoproj/alert-manager/blob/d73e1189ad3dc0b2f1a6142aa28837ebce145e3d/controllers/alertsconfig_controller.go#L173) and [update](https://github.com/keikoproj/alert-manager/blob/d73e1189ad3dc0b2f1a6142aa28837ebce145e3d/controllers/alertsconfig_controller.go#L203)  , please go through the code to understand the logic.